src/dailyNoteAppendCommand.tsx

```diff
@@ -0,0 +1,73 @@
+import { getPreferenceValues, open, showToast, Toast } from "@raycast/api";
+import { URLSearchParams } from "url";
+import { applyTemplates, loadObsidianJson, vaultPluginCheck } from "./utils/utils";
+
+interface DailyNoteAppendArgs {
+  text: string;
+}
+
+interface Preferences {
+  appendTemplate?: string;
+  vaultName?: string;
+  heading?: string;
+}
+
+export default async function DailyNoteAppend(props: { arguments: DailyNoteAppendArgs }) {
+  const { text } = props.arguments;
+  const { appendTemplate, heading, vaultName } = getPreferenceValues<Preferences>();
+
+  const vaults = await loadObsidianJson();
```

Comment:
Can't we use `useObsidianVaults()` from `utils.tsx` here? This would make it more consistent with the other commands which all use the `useObsidianVaults()` function.

Reply:
Unfortunately we can't use hooks like `useObsidianVaults` here because I've defined this as a `no-view` command. This means that this function is only called once, more like a script, as opposed to a react component function that is called once per render.

I don't think this command warrants a UI, so `no-view` seemed appropriate. That being said, if you prefer the consistency of being able to use hooks, and the opportunity to reuse more code, I could rework this into a `view` command. Changing it to a `view` would also allow us to do the vault selection as you suggest below. What do you think?

In the case where the append succeeds, I think the view would just show a success message like "Append to Daily Note in vaultName vault succeeded!"

Reply:
Actually in the success case I can pop the navigation and close the window like the daily notes command does 👍🏻